src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -100,8 +100,10 @@ private GeneratedMethod BuildTypeNoCache(ServiceCallSite callSite)
                 owner: GetType(),
                 skipVisibility: true);
 
-            ILEmitCallSiteAnalysisResult info = ILEmitCallSiteAnalyzer.Instance.CollectGenerationInfo(callSite);
-            ILGenerator ilGenerator = dynamicMethod.GetILGenerator(info.Size);
+            // In traces we've seen methods range from 100B - 4K sized methods since we've
+            // stop trying to inline everything into scoped methods. We'll pay for a couple of resizes
+            // so there'll be allocations but we could potentially change ILGenerator to use the array pool
```

Comment:
> but we could potentially change ILGenerator to use the array pool

I'm skeptical we would do this.
https://github.com/dotnet/designs/blob/main/accepted/2021/runtime-security-mitigations.md#protect-intermediate-language-il-code
It introduces potential issues where an array is erronously returned to the pool while still in use, then the ILGenerator picks up that array and uses it to write IL, and now whoever returned the array to the pool is potentially able to write IL that'll be executed.

Reply:
Sure, I don't think it matters anymore really. We don't generate large methods anymore anyways 😄. Also now that we have this event, we can ask people to turn it on and see what sizes we get in practice to adjust this number.